core/adminserver/admin_server.go

```diff
@@ -370,6 +389,39 @@ func (s *Server) UndeleteDirectory(ctx context.Context, in *pb.UndeleteDirectory
 	return nil, status.Errorf(codes.Unimplemented, "not implemented")
 }
 
+// ListInputLogs returns a list of input logs for a directory.
+func (s *Server) ListInputLogs(ctx context.Context, in *pb.ListInputLogsRequest) (*pb.ListInputLogsResponse, error) {
+	logIDs, err := s.logsAdmin.ListLogs(ctx, in.GetDirectoryId(), in.GetFilterWritable())
+	if err != nil {
+		s := status.Convert(err)
+		return nil, status.Errorf(s.Code(), "adminserver: ListLogs(): %v", s.Message())
+	}
+	inputLogs := make([]*pb.InputLog, 0, len(logIDs))
```

Comment:
I don't know how much better this is than:
inputLogs := []*pb.InputLog{}
which I find more readable

Reply:
The only difference is that the way it's written pre-allocates memory so we can avoid multiple runtime array resizing.